src/schema_plus/core.cljc

```diff
@@ -203,42 +204,12 @@
                 (assoc m# ::incomplete true)))
 
              ; define function to finish building instances
-             (defn ~build-fn-name
-               [this#]
-               (let [final# (dissoc this# ::incomplete)]
-                 (s/validate schema-obj# final#)
-                 final#)))
-
+             (let [build-fn# (defn ~build-fn-name
+                               [this#]
+                               (let [final# (dissoc this# ::incomplete)]
+                                 (s/validate schema-obj# final#)
+                                 final#))]
+              (defmacro ~build-macro-name
+                ([~'& forms#]
+                  `(~build-fn# (dissoc (-> ~@forms#) ::incomplete))))))
```

Comment:
The build-fn already handles the dissoc, so that doesn't need to be duplicated here, right?